apps/webservice/src/app/[workspaceSlug]/_components/DeleteDeployment.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useParams, useRouter } from "next/navigation";
+import { z } from "zod";
+
+import {
+  AlertDialog,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@ctrlplane/ui/alert-dialog";
+import { Button } from "@ctrlplane/ui/button";
+
+import { api } from "~/trpc/react";
+
+const deleteDeploymentSchema = z.object({
+  deploymentId: z.string().uuid(),
+});
+
+type DeleteDeploymentProps = {
+  deploymentId: string;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+};
+
+export const DeleteDeploymentDialog: React.FC<DeleteDeploymentProps> = ({
+  deploymentId,
+  isOpen,
+  setIsOpen,
+}) => {
+  const { workspaceSlug } = useParams<{ workspaceSlug: string }>();
+  const router = useRouter();
+  const deleteDeployment = api.deployment.delete.useMutation();
+  const utils = api.useUtils();
+
+  const onDelete = async () => {
+    const validation = deleteDeploymentSchema.safeParse({ deploymentId });
+
+    if (!validation.success) {
+      console.error("Invalid deployment ID:", deploymentId);
+      return;
+    }
+    await deleteDeployment.mutateAsync(deploymentId);
```

Comment:
```suggestion
    await deleteDeployment.mutateAsync(deploymentId);
```

The backend will throw an error, we don't need to handle this client side